_episodes/script.md

```diff
@@ -1,38 +1,218 @@
-  - distribute as seperate package via pip / cran / ...
+__include and run their own script in a step at runtime__
+
+> ## Exercise 1:
+>
+> Complete the `requirements` field for the `CommandLineTool` that creates a script at runtime. As a HINT, `DockerRequirement` has already been filled-in.
+>
+> ~~~
+> class: CommandLineTool
+> cwlVersion: v1.1
+> requirements:
+>   DockerRequirement:
+>     dockerPull: 'debian:stable'
+>   -----------------:
+> ~~~
+> {: .language-yaml }
+>
+> > ## Solution
+> >
+> > ~~~
+> > class: CommandLineTool
+> > cwlVersion: v1.1
+> > requirements:
+> >    DockerRequirement:
+> >      dockerPull: 'debian:stable'
+> >    InitialWorkDirRequirement:
+> > ~~~
+> > {: .language-yaml }
+> >
+> {: .solution}
+{: .challenge}
+
+> ## Exercise 2:
+>
+> Identify `CWL` keywords for defining script name `script.sh` and the contents of this script under `InitialWorkDirRequirement`.
+>
+> ~~~
+> InitialWorkDirRequirement:
+>   listing:
+>     - ------: script.sh
+>       ------: |
+>         echo "*Documenting input*" && \
+>         echo "Input received: $(inputs.message)" && \
+>         echo "Exit"
+>
+> inputs:
+>   message:
+>     type: string
+> ~~~
+> {: .language-yaml }
+>
+> > ## Solution:
+> >
+> > ~~~
+> > InitialWorkDirRequirement:
+> >   listing:
+> >     - entryname: script.sh
+> >       entry: |
+> >         echo "*Documenting input*" && \
+> >         echo "Input received: $(inputs.message)" && \
+> >         echo "Exit"
+> >
+> > inputs:
+> >   message:
+> >     type: string
+> > ~~~
+> > {: .language-yaml }
+> >
+> {: .solution}
+{: .challenge}
+
+> ## Exercise 3:
+>
+> Since we are using `echo` in the script (as shown below) - what is the appropriate `type` in the `outputs` section of following code block to capture standard output?
+>
+> ~~~
+> class: CommandLineTool
+> cwlVersion: v1.1
+> requirements:
+>   DockerRequirement:
+>     dockerPull: 'debian:stable'
+>   InitialWorkDirRequirement:
+>     listing:
+>       - entryname: script.sh
+>         entry: |
+>           echo "*Documenting input*" && \
+>           echo "Input received: $(inputs.message)" && \
+>           echo "Exit"
+>
+> inputs:
+>   message:
+>     type: string
+>
+> stdout: "message.txt"
+>
+> outputs:
+>  message:
+>    type: ----
+> ~~~
+> {: .language-yaml }
+>
+> Your options are:
+> A. File
+> B. Directory
+> C. stdout
+> D. string
+>
+> > ## Solution:
+> >
+> > C. stdout
+> >
+> {: .solution}
+{: .challenge}
+
+> ## Exercise 4:
+>
+> Fix the `baseCommand` in following code snippet to execute the script we have created in previous exercises.
+>
+> ~~~
+> baseCommand: []
+> ~~~
+> {: .language-yaml }
+>
+> > ## Solution:
+> >
+> > ~~~
+> > baseCommand: ["sh", "script.sh"]
+> > ~~~
+> > {: .language-yaml }
+> >
+> {: .solution}
+{: .challenge}
+
+> ## Exercise 5:
+>
+> CHALLENGE question. Extend the `outputs` section of the following CWLtool definition to capture the script we have created along with tools' standard output.
```

Comment:
```suggestion
> CHALLENGE question. Extend the `outputs` section of the following CWLtool definition to capture the script we have created along with tools' standard output.
> 
> This will help you inspect the generated script and is useful in more complex situations to troubleshoot related issues.
```

Reply:
Thank you @mr-c for the great suggestions.